src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -806,41 +636,19 @@ DistributedObjectMetadataSyncCommandList(void)
 
 
 /*
- * GetDistributedTableDDLEvents returns the full set of DDL commands necessary to
- * create the given distributed table on a worker. The list includes setting up any
- * sequences, setting the owner of the table, inserting table and shard metadata,
- * setting the truncate trigger and foreign key constraints.
+ * GetDistributedTableMetadataEvents returns the full set of DDL commands necessary to
+ * create the given distributed table metadata on a worker. The list includes setting up
+ * any shard metadata, setting the truncate trigger and foreign key constraints.
  */
 static List *
-GetDistributedTableDDLEvents(Oid relationId)
+GetDistributedTableMetadataEvents(Oid relationId)
 {
 	CitusTableCacheEntry *cacheEntry = GetCitusTableCacheEntry(relationId);
 
 	List *commandList = NIL;
-	IncludeSequenceDefaults includeSequenceDefaults = WORKER_NEXTVAL_SEQUENCE_DEFAULTS;
 
 	/* if the table is owned by an extension we only propagate pg_dist_* records */
 	bool tableOwnedByExtension = IsTableOwnedByExtension(relationId);
-	if (!tableOwnedByExtension)
-	{
-		/*
-		 * Commands to create the table, these commands are TableDDLCommands so lets
-		 * materialize to the non-sharded version
-		 */
-		List *tableDDLCommands = GetFullTableCreationCommands(relationId,
-															  includeSequenceDefaults);
-		TableDDLCommand *tableDDLCommand = NULL;
-		foreach_ptr(tableDDLCommand, tableDDLCommands)
-		{
-			Assert(CitusIsA(tableDDLCommand, TableDDLCommand));
-			commandList = lappend(commandList, GetTableDDLCommand(tableDDLCommand));
-		}
-
-		/* command to associate sequences with table */
-		List *sequenceDependencyCommandList = SequenceDependencyCommandList(
-			relationId);
-		commandList = list_concat(commandList, sequenceDependencyCommandList);
-	}
```

Comment:
`GetDistributedTableMetadataEvents` does multiple things. Creates pg_dist* entries and truncate trigger/partitiioning/foreiign keys. Can yoo eitther rename or split the functiio into two?

Reply:
I've kept only pg_dist_* entries and truncate trigger. I've kept truncate trigger because it should be created even if the table is owned by extension and it's purpose is dropping metadata entries.